app/models/game.rb

```diff
@@ -5,16 +5,55 @@ class Game < ApplicationRecord
 
   validates :name, presence: true
   after_create :populate_board
-	
+  
+  
+  def stalemate?(color)
+    current_pieces = friendly_pieces(color)
+    possible_moves = []
+    current_pieces.each do |piece|
+      8.times do |x|
+        8.times do |y|
+          possible_moves << [x, y] if piece.valid_move?(x, y)
+        end
```

Comment:
looks great but i think it should be piece.valid_move && !put_in_check

Reply:
Yeah I was trying that initially but put_in_check didn't seem to be working so I left it off until issues are resolved

Reply:
I think the implementation of in_check? In put_in_check is off. In_check? Should take a argument

Reply:
yep, it should take a king as the argument. I know check is broken at the moment. Zach mentioned you might have a fix @chris-campbell? If so, throw up a pr with that logic and I'd be happy to test it. If not I'll snap to it to get the check logic working. I'm sorry I didn't at the time- tests weren't working so it was a hail mary method.

Reply:
Check has minor issues I believe. I had sent a possible solution a couple of days ago on slack but i think it was sent privately. Im not sure but my slack has been acting up had to reinstall. but if we can go over this tonight on our meet up if you'd like.

Reply:
put_in_check? also uses a king object that isn't declared or passed in as an argument in the function.
example: current_state = king.where(color: king.color).in_check?

I expect a undefined method error for "king.where" and "king.color"

**Something like:**
_king = pieces.find_by(type: 'King', color: 'white')_ 
should return a king object to work with.

Also I think rather then king.where(), it should be something like pieces.where().